### PR TITLE
[113] wip

### PR DESCRIPTION
**Pull request by connectbo**
This pr adds a new check for the length of the resource name. I set it to be 63 characters according to the issue itself, happy to change. Also made our resource table more bulletproof when the names are really long.

image

On narrower mobile view:

image

### CODE REVIEW

File: src/data/checkfiles/default_checkfiles/validate-resc-name-length.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import { Link } from '@reach/router'
+
+export const validateRescNameLength = {
+    "name": "Each resource has a valid name length",
+    "description": `Checks if the length of each resource is within 63 characters.`,
+    "min_server_version": "4.2.0",
+    "max_server_version": "4.2.10",
```
Comment (trel) [inline src/data/checkfiles/default_checkfiles/validate-resc-name-length.js:8]: 4.2.11 is now released.

**Review by trel — COMMENTED** [review]
looks pretty good - still some weird 400 things on hitting save... seems related to our ongoing other weird stuff in my setup.

File: src/data/checkfiles/default_checkfiles/validate-resc-name-length.js (hunk 1 above)
Comment (connectbo) [reply src/data/checkfiles/default_checkfiles/validate-resc-name-length.js:8]: I changed both `min_server_version` and `max_server_version` to be empty string in all check files for now so zmt won't check it unless it's specified.

Comment (trel) [reply src/data/checkfiles/default_checkfiles/validate-resc-name-length.js:8]: both are now optional for a checkfile - yes.

but we want to define the ones we ship with minimums of 4.2.0 for now please.

**Comment by trel** [comment]
I think this is ready.   please # them both.

I'm still getting some strange behavior if selecting a resource to edit... making the name 'too long'... and then hitting enter or clicking the pencil.   it says failure... but then doing it again... flashes green, but doesn't actually update.    

but we can chase it down later i think - it's only *after* a failure does the state seem a bit weird.   a page refresh makes it fine again.

**Comment by connectbo** [comment]
I see. Yes, I can reproduce that error, should be something related to the event handler. I will address that later. added #

**Pull request merged**